modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -47,8 +48,34 @@ func (*Keeper) Acknowledgement(ctx context.Context, acknowledgement *channeltype
 }
 
 // RecvPacket implements the PacketMsgServer RecvPacket method.
-func (*Keeper) RecvPacket(ctx context.Context, packet *channeltypesv2.MsgRecvPacket) (*channeltypesv2.MsgRecvPacketResponse, error) {
-	panic("implement me")
+func (k *Keeper) RecvPacket(ctx context.Context, msg *channeltypesv2.MsgRecvPacket) (*channeltypesv2.MsgRecvPacketResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	packet := msg.Packet
+	err := k.recvPacket(ctx, packet, msg.ProofCommitment, msg.ProofHeight)
+	if err != nil {
+		sdkCtx.Logger().Error("send packet failed", "source-id", packet.SourceId, "error", errorsmod.Wrap(err, "send packet failed"))
+		return nil, errorsmod.Wrapf(err, "send packet failed for source id: %s", packet.SourceId)
+	}
+
+	signer, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		sdkCtx.Logger().Error("send packet failed", "error", errorsmod.Wrap(err, "invalid address for msg Signer"))
+		return nil, errorsmod.Wrap(err, "invalid address for msg Signer")
+	}
+
+	_ = signer
+
+	// TODO: implement once app router is wired up.
+	// https://github.com/cosmos/ibc-go/issues/7384
+	// for _, pd := range packet.PacketData {
+	//	cbs := k.PortKeeper.AppRouter.Route(pd.SourcePort)
+	//	err := cbs.OnRecvPacket(ctx, packet, msg.ProofCommitment, msg.ProofHeight, signer)
+	//	if err != nil {
+	//		return nil, err
+	//	}
+	// }
+
```

Comment:
worth mentioning that the remainder of this fn should be wired up in https://github.com/cosmos/ibc-go/issues/7354